### PR TITLE
`ActorContext` in `CommandContext`

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
This PR brings `ActorContext` into `CommandContext`.

`ActorContext` now belongs to the `base` package.

Other changes:
* `CommandEnvelope` was extended for obtaining `TenantId`.
* The short link for `FieldMask` specs are used in Javadocs.

### CODE REVIEW

**Comment by alexander-yevsyukov** [comment]
@armiol, PTAL.

File: client/src/main/java/org/spine3/base/Commands.java [hunk 1]
```diff
@@ -161,14 +163,19 @@ public static CommandContext createContext(@Nullable TenantId tenantId,
...
      */
-    public static CommandContext newContextBasedOn(CommandContext commandContext) {
-        checkNotNull(commandContext);
-        final CommandContext.Builder result = commandContext.toBuilder()
-                                                            .setCommandId(generateId())
-                                                            .setTimestamp(getCurrentTime());
+    public static CommandContext newContextBasedOn(CommandContext value) {
+        checkNotNull(value);
+        final ActorContext.Builder withCurrentTime =
+                value.getActorContext()
```
Comment (armiol) [inline client/src/main/java/org/spine3/base/Commands.java]: This expression may start on a previous line.

File: client/src/main/java/org/spine3/base/Commands.java [hunk 2]
```diff
@@ -161,14 +163,19 @@ public static CommandContext createContext(@Nullable TenantId tenantId,
...
+                     .toBuilder()
+                     .setTimestamp(getCurrentTime());
+        final CommandContext.Builder result =
+                value.toBuilder()
```
Comment (armiol) [inline client/src/main/java/org/spine3/base/Commands.java]: This expression may start on a previous line.

File: server/src/test/java/org/spine3/server/commandbus/AbstractCommandBusTestSuite.java [hunk 3]
```diff
@@ -116,11 +117,15 @@ void checkCommandError(Throwable throwable,
...
-        final Command invalidCmd = cmd.toBuilder()
-                                      .setContext(cmd.getContext()
-                                                     .toBuilder()
-                                                     .setTenantId(TenantId.getDefaultInstance()))
-                                      .build();
+        final ActorContext.Builder withNoTenant =
+                ActorContext.newBuilder()
+                            .setTenantId(TenantId.getDefaultInstance());
+        final Command invalidCmd =
+                cmd.toBuilder()
```
Comment (armiol) [inline server/src/test/java/org/spine3/server/commandbus/AbstractCommandBusTestSuite.java]: This expression may start on the previous line.

File: client/src/main/proto/spine/client/query.proto [hunk 4]
```diff
@@ -65,7 +66,7 @@ message Query {
     google.protobuf.FieldMask field_mask = 3;
 
     // Service information about the environment in which the query was created.
-    ActorContext context = 4 [(required) = true, (valid) = true,
+    base.ActorContext context = 4 [(required) = true, (valid) = true,
                               (if_invalid).msg_format = "Invalid actor context"];
```
Comment (armiol) [inline client/src/main/proto/spine/client/query.proto]: Please adjust the formatting.

File: client/src/main/proto/spine/base/command.proto [hunk 5]
```diff
@@ -72,20 +70,8 @@ message CommandContext {
...
-
-    // The user who created the command.
-    users.UserId actor = 3 [(required) = true];
-
-    // The time when the command was created.
-    google.protobuf.Timestamp timestamp = 4 [(required) = true];
-
-    // The zone offset from which the command is made.
-    time.ZoneOffset zone_offset = 5 [(required) = true];
+    // Information about the user who created the command.
```
Comment (armiol) [inline client/src/main/proto/spine/base/command.proto]: As long as the `ActorContext` has timestamp and zone offset, that's not all `Information about the user`.

**Pull request merged**